pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -415,7 +415,7 @@ type KubeletConfiguration struct {
 	// enable gathering custom metrics.
 	EnableCustomMetrics bool `json:"enableCustomMetrics"`
 	// Comma-delimited list of hard eviction expressions.  For example, 'memory.available<300Mi'.
-	EvictionHard string `json:"evictionHard"`
+	EvictionHard *string `json:"evictionHard"`
```

Comment:
I think there's no reason to have the internal & external versions differ for the time being. Also, this should have `omitempty`.